packages/cli/validate-macos-version.ts

```diff
@@ -0,0 +1,67 @@
+import os from "node:os";
+
+/**
+ * Validates that the current macOS version supports workerd.
+ * Throws an error if running on macOS below version 13.5.
+ */
+export function validateMacOSVersion(): void {
+	if (process.platform !== "darwin") {
+		return;
+	}
+
+	if (process.env.CI === "true") {
+		return;
+	}
+
+	const release = os.release();
+	const macOSVersion = darwinVersionToMacOSVersion(release);
+
+	if (macOSVersion && isVersionLessThan(macOSVersion, "13.5.0")) {
+		throw new Error(
+			`Unsupported macOS version: We don't support the current version of macOS (${macOSVersion}). ` +
+				`The minimum requirement is macOS 13.5+. ` +
+				`If you cannot upgrade your version of macOS, you could try running create-cloudflare/wrangler in a DevContainer setup with Linux.`
+		);
```

Comment:
Let's give a slightly more informational error message about why we don't support older Macos...

```suggestion
		throw new Error(
			`Unsupported macOS version: The Cloudflare Workers runtime cannot run on the current version of macOS (${macOSVersion}). ` +
				`The minimum requirement is macOS 13.5+. See https://github.com/cloudflare/workerd?tab=readme-ov-file#running-workerd` +
				`If you cannot upgrade your version of macOS, you could try running create-cloudflare/wrangler in a DevContainer setup with Linux.`
		);
```